RUFAS/routines/animal/purchased_feed_emissions_estimator.py

```diff
@@ -0,0 +1,161 @@
+            if feed_id == "dry_matter_intake_total":
+                continue
+            if feed_id in self.missing_feed_ids:
+                continue
+            if feed_id not in self.feed_emissions.keys():
+                om.add_warning(
+                    "Missing Purchased Feed Emissions",
+                    f"Missing purchased feed emissions data for RuFaS feed {feed_id}.",
+                    info_map
+                )
+                self.missing_feed_ids.append(feed_id)
+                continue
+            emissions = amount_fed * self.feed_emissions[feed_id]
+            emissions_per_feed_id["feed_emissions_total"] += emissions
+            emissions_per_feed_id[feed_id] = emissions
+        return emissions_per_feed_id
+
+    def _get_geographic_coordinates(self) -> (float, float):
+        info_map = {
+            "class": self.__class__.__name__,
+            "function": self._get_geographic_coordinates.__name__
+        }
+
+        madison_wi_latitude = 43.073
+        madison_wi_longitude = -89.401
+
+        field_keys: list[str] = im.get_data_keys_by_properties("field_properties")
+
+        if not field_keys:
+            warning_name = "Default Feed Emissions data"
+            warning_message = "Could not obtain feed emissions geographic attributes, defaulting to Madison, WI."
+            om.add_warning(warning_name, warning_message, info_map)
+
+            return {"latitude": madison_wi_latitude, "longitude": madison_wi_longitude}
+
+        latitude = im.get_data(f"{field_keys[0]}.absolute_latitude")
+
+        longitude = im.get_data(f"{field_keys[0]}.longitude")
+
+        longitude = abs(longitude) * -1
+
+        return latitude, longitude
+
+    def _get_county_code(self, latitude: float, longitude: float) -> int:
+        info_map = {
+            "class": self.__class__,
+            "function": self._get_county_code.__name__
+        }
+
+        endpoint = "https://geo.fcc.gov/api/census/block/find"
+        params = {"latitude": latitude, "longitude": longitude, "format": "json"}
+
+        attempts = 3
+        for attempt_count in range(attempts):
+            om.add_log(
+                "Feed Emissions API Call",
+                f"Calling the FCC's FIPS County Code API, attempt: {attempt_count}",
+                info_map
+            )
+            response = requests.get(endpoint, params=params)
+            if response.status_code == 200:
+                answer = response.json()
+                om.add_log(
+                    "Successful Feed Emissions API Call",
+                    f"Got successful response from the FCC's FIPS County Code API on attempt: {attempt_count}",
+                    info_map
+                )
+                return int(answer["County"]["FIPS"])
+            om.add_error(
+                "Error Feed Emissions API Call",
+                f"FCC's FIPS County Code API attempt: {attempt_count} failed with status code: {response.status_code}.",

```

Comment:
```suggestion
                f"FCC's FIPS County Code API {attempt_count=} failed with {response.status_code=}.",

```
you can shorten this f-strings like so.